modulus/utils/filesystem.py

```diff
@@ -43,11 +46,88 @@ def _get_fs(path):
         return fsspec.filesystem("file")
 
 
-def _download_cached(path: str, recursive: bool = False) -> str:
+def _download_ngc_model_file(path: str, out_path: str, timeout: int = 300) -> str:
+    """Pulls files from model registry on NGC. Supports private registries when NGC
+    API key is set the the API_KEY enviroment variable. If download file is a zip folder
+    it will get unzipped.
+
+    Args:
+        path (str): NGC model file path of form `ngc://org/team/model/version/filename`
+        out_path (str): Output path to save file / folder as
+        timeout (int): Time out of requests, default 5 minutes
+
+    Raises:
+        ValueError: Invlaid url
+
+    Returns:
+        str: output file / folder path
+    """
+    # Strip ngc url prefix
+    if not path.startswith("ngc://"):
+        raise ValueError(
+            "Invalid URL, should be of form ngc://org/team/model/version/filename"
+        )
+    path = path[6:]
+
+    if len(path.split("/")) != 5:
+        raise ValueError(
+            "Invalid URL, should be of form ngc://org/team/model/version/filename"
+        )
+
+    (org, team, model, version, filename) = path.split("/")
+    token = ""
+    # If API key environment variable
+    if "API_KEY" in os.environ:
+        try:
+            session = requests.Session()
+            session.auth = ("$oauthtoken", os.environ["API_KEY"])
+            headers = {"Accept": "application/json"}
+            authn_url = f"https://authn.nvidia.com/token?service=ngc&scope=group/ngc:{org}&group/ngc:{org}/{team}"
+            r = session.get(authn_url, headers=headers, timeout=5)
+            r.raise_for_status()
+            token = json.loads(r.content)["token"]
+        except requests.exceptions.RequestException:
+            logger.warning("Failed to get JWT using the API set in API_KEY")
```

Comment:
`set in API_KEY` -> `set in the API_KEY environment variable`?

Reply:
Also, good practice to re-raise the exception.
```python
try:
    ...
except requests.exceptions.RequestException:
    logger.warning("Failed to get JWT using the API set in API_KEY")
    raise  # Will re-raise the exception with the same stack trace
```

Reply:
Sounds good, I had originally left this to get passed so if someone was actually trying to access a public file it would still make the attempt despite the auth failure.

But I agree that it should just fail here if someone has the environment variable set so its transparent.

Reply:
Updated!